### PR TITLE
Frienshipv1

### PR DESCRIPTION
**Pull request by angelacuahutle**
Dear TSE

I have covered Milestone 5: Friendships v1.

For this task, I have created a model with associations and all requested features for friendships.
I also included their respective unit and integration test.

Thanks

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @angelacuahutle, 

**This is a great project you have going on, well-done lad**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️

- [x] Enable the GitHub Actions in the Actions tab. This will ensure that your linter checks are running.
- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from Pathwright). Currently, you app shows all posts from all users even when they are not friends.

_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: app/models/user.rb [hunk 1]
```diff
@@ -9,4 +9,27 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friends, through: :friendship
+  has_many :friends, through: :friendships_confirmation
+  has_many :friend_requests, through: :reverse_friendships, source: :user
```
Comment (mke2111) [inline app/models/user.rb]: - [x] The friendship feature has not been fully implemented. Currently, a user can press the Add friend button but nothing happens except leading to an empty page. A user should be able to accept or reject friend requests.
- [x] There should be a button to add friend on user's single page.
- [x] A user should be able to see all friend requests sent their way.

File: spec/models/friendship_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Friendship, type: :model do
+  describe 'associations' do
+    it { should belong_to(:user).class_name('User') }
```
Comment (mke2111) [inline spec/models/friendship_spec.rb:5]: - [x] Add Unit and Integration tests for your app. Ensure to cater for all models

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @angelacuahutle, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from Pathwright). Currently, your app shows all posts from all users even when they are not friends.
- [x] Add Unit and Integration tests for your app. Ensure to cater for all models. Currently, you only have 2 tests(unit tests) for your entire app.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @angelacuahutle , 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**